src/v/cloud_topics/level_zero/gc/level_zero_gc.h

```diff
@@ -180,6 +181,23 @@ class level_zero_gc {
      */
     class epoch_source {
     public:
+        struct partitions_snapshot {
+            using partition_map = chunked_hash_map<
+              model::topic_namespace,
+              chunked_vector<model::partition_id>,
+              model::topic_namespace_hash,
+              model::topic_namespace_eq>;
+
+            partition_map partitions;
+            model::revision_id last_applied;
```

Comment:
nit: since we're in the epoch_source class here, does it make conceptual sense to cast this to an epoch?